business_requirement/models/business.py

```diff
@@ -290,6 +290,17 @@ def write(self, vals):
                     raise ValidationError(_(
                         'You can only move to the following stage: '
                         'draft/confirmed /cancel/drop.'))
+            if vals['state'] == 'stakeholder_approval':
```

Comment:
Not sure why you have this modification here

Reply:
In #259 
>  Approval date should be when the SOA state is changed

That's why I have added this code here